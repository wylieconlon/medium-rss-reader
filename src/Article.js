import React, { Component } from 'react';

class Article extends Component {
  render() {
    const article = this.props.article;
    const isExpanded = this.props.isExpanded;

    return (<article className={`${isExpanded ? 'is-expanded' : ''}`}>
      <h2><a href={article.link}>{article.title}</a></h2>
      <p>By {article.creator}</p>

      {!isExpanded && <p dangerouslySetInnerHTML={article.description} />}

      {!isExpanded && article.hasFullArticle &&
        <button className="expander"
          onClick={this.props.onExpand.bind(this, article)}
        >
          Expand article
        </button>
      }

      {isExpanded && <div dangerouslySetInnerHTML={article.content} />}

      <footer>
        Published <time dateTime={article.publicationDateString}>
          {article.publicationDateFormatted}
        </time>
        &nbsp; &bull; Updated at <time dateTime={article.updatedDateString}>
          {article.updatedDateFormatted}
        </time>
      </footer>
    </article>);
  }
}

export default Article;
