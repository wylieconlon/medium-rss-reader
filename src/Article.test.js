import React from 'react';
import { shallow } from 'enzyme';
import Article from './Article';

it('renders without crashing', () => {
  expect(shallow(<Article />).html()).toMatchSnapshot();
});

it('renders a more full article', () => {
  const dateFormatted = '10/17/18';
  const article = {
    title: 'Fake Article',
    description: {__html: '<p>Some articles have a preview here</p>'},
    hasFullArticle: true,
    content: {__html: '<p>This is the full content</p>'},
    link: 'http://medium.com',
    categories: ['technology', 'politics'],
    shortLink: 'unique key',
    creator: 'Wylie Conlon',
    publicationDateString: dateFormatted,
    updatedDateString: dateFormatted,
    publicationDateFormatted: dateFormatted,
    updatedDateFormatted: dateFormatted,
  };

  const isExpanded = false;

  expect(shallow(<Article
    article={article}
    isExpanded={isExpanded}
    onExpand={jest.fn()}
  />).html()).toMatchSnapshot();
});
