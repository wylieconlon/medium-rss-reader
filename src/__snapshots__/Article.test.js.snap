// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a more full article 1`] = `"<article class=\\"\\"><h2><a href=\\"http://medium.com\\">Fake Article</a></h2><p>By Wylie Conlon</p><p><p>Some articles have a preview here</p></p><button class=\\"expander\\">Expand article</button><footer>Published <time dateTime=\\"10/17/18\\">10/17/18</time>  • Updated at <time dateTime=\\"10/17/18\\">10/17/18</time></footer></article>"`;

exports[`renders without crashing 1`] = `null`;
